### DISCUSSION TITLE
Installed Nextcloud AIO on RHEL 9 successfully + two short questions + one suggestion

### DISCUSSION THREAD

**Discussion by Verhoeckx** [nextcloud/all-in-one#1762] [Questions] [answered]
Good news: 
I successfully installed Nextcloud AIO on RHEL 9! In the coming weeks I will finish my blog post about it and post it here!

Two short questions:
1. Do I understand it right and is adding `NEXTCLOUD_DATADIR` not necessary because all the files and folders are stored in `/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/`?
2. As I understand the backup directory (default: `/mnt/backup`) doesn't need any special file permissions?

One suggestion:
Add the option `-d` to the docker run command so that it's started as a background process. This way the command line stays free for other commands.

**Comment by szaimen** [comment, marked as answer]
> Good news: I successfully installed Nextcloud AIO on RHEL 9! In the coming weeks I will finish my blog post about it and post it here!

Great!

> 1. Do I understand it right and is adding `NEXTCLOUD_DATADIR` not necessary because all the files and folders are stored in `/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/`?

That is correct 👍 (if `NEXTCLOUD_DATADIR` is not given, it will store the files in the mentioned directory.)

> 2. As I understand the backup directory (default: `/mnt/backup`) doesn't need any special file permissions?

That is correct, as well 👍 (at least as long as you don't use docker rootless).

> One suggestion: Add the option `-d` to the docker run command so that it's started as a background process. This way the command line stays free for other commands.

Thanks for the suggestion. However since the docker container does a few things on startup and also shows a hint how to proceed, we will keep it like it is currently. But you are obviously free to add the -d option yourself in your guide.

**Comment by Verhoeckx** [comment]
> That is correct +1 (if NEXTCLOUD_DATADIR is not given, it will store the files in the mentioned directory.)

'Funny', I always thought that a volume was a single file, but in this case it's just a mounted directory?

> Thanks for the suggestion. However since the docker container does a few things on startup and also shows a hint how to proceed, we will keep it like it is currently. But you are obviously free to add the -d option yourself in your guide.

Oh, yes, you are right, there is indeed a hint shown where to open the Nextcloud AIO interface. Understandable that you keep it the way it is! And yes, in my documentation I will add the -d option.

Thanks for answering!

**Reply by szaimen** [reply]
> > That is correct +1 (if NEXTCLOUD_DATADIR is not given, it will store the files in the mentioned directory.)
> 
> 'Funny', I always thought that a volume was a single file, but in this case it's just a mounted directory?

Docker volumes are usually directories :) See https://docs.docker.com/storage/volumes/

> > Thanks for the suggestion. However since the docker container does a few things on startup and also shows a hint how to proceed, we will keep it like it is currently. But you are obviously free to add the -d option yourself in your guide.
> 
> Oh, yes, you are right, there is indeed a hint shown where to open the Nextcloud AIO interface. Understandable that you keep it the way it is! And yes, in my documentation I will add the -d option.
> 
> Thanks for answering!

👍

**Comment by Verhoeckx** [comment]
> Docker volumes are usually directories :) See https://docs.docker.com/storage/volumes/

I read (scanned) that page before, but what I find so frustrating, is that it does describe the advantages of using a volume, but it's doesn't say what it actually is! This happens so often in documentation :disappointed: !

**Reply by szaimen** [reply]
Maybe this helps? https://docs.docker.com/storage/#more-details-about-mount-types

**Comment by Verhoeckx** [comment]
That documentation is indeed better!

**Volumes are stored in a part of the host filesystem which is managed by Docker** (/var/lib/docker/volumes/ on Linux). Non-Docker processes should not modify this part of the filesystem.

When you create a volume, it is stored within a directory on the Docker host. When you mount the volume into a container, **this directory** is what is mounted into the container. This is similar to the way that bind mounts work, except that volumes are managed by Docker and are isolated from the core functionality of the host machine.

Why don't they say so :smile: !

**Comment by nogjam** [comment]
@Verhoeckx Did you end up finishing that blog article? That would be helpful. :)

**Comment by Verhoeckx** [comment]
@n8jhj: Yes, I did finish that blog post!

You can find it here:
[https://jeroenverhoeckx.com/install-nextcloud-with-docker.html](https://jeroenverhoeckx.com/install-nextcloud-with-docker.html)

_Hope it's of some use to you!_

**Reply by nogjam** [reply]
Thank you!
